core/src/main/java/org/jdbi/v3/core/statement/SqlLoggable.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.statement;
+
+import java.sql.SQLException;
+
+@FunctionalInterface
+public interface SqlLoggable<T> {
+    T invoke() throws SQLException;
```

Comment:
SQLException is a good compromise. :+1:

Reply:
well I can leave it as that or make it X with an unsafe cast. I see pros and cons to both, what you?

Reply:
If I make it X though, SqlLogger will also become generic with an exception type... maybe not desirable

Reply:
I think `SQLException` is reasonable--let's just stick with that.

Reply:
Yeah, I just implemented and pushed a refactor to X for demonstration. It works but I don't see it adding any value, and it's a burden on implementers of `logException`. I'd stick to SQLException too.